tests/client/css_variables_tests.cpp

```diff
@@ -0,0 +1,251 @@
+{
+  SECTION("Parse CSS custom property declaration")
+  {
+    CSSStyleDeclaration style("--main-color: red;");
+
+    // Check if custom property is stored
+    REQUIRE(style.getPropertyValue("--main-color") == "red");
+    REQUIRE(style.length() == 1);
+    REQUIRE(style.item(0) == "--main-color");
+  }
+
+  SECTION("CSS custom property names are case sensitive")
+  {
+    CSSStyleDeclaration style("--Main-Color: blue; --main-color: red;");
+
+    REQUIRE(style.getPropertyValue("--Main-Color") == "blue");
+    REQUIRE(style.getPropertyValue("--main-color") == "red");
+    REQUIRE(style.getPropertyValue("--MAIN-COLOR") == "");
+    REQUIRE(style.length() == 2);
+  }
+
+  SECTION("CSS custom property with var() function")
+  {
+    CSSStyleDeclaration style("--main-color: red; color: var(--main-color);");
+
+    REQUIRE(style.getPropertyValue("--main-color") == "red");
+
+    // The raw value should contain var() until resolved
+    std::string colorValue = style.getPropertyValue("color");
+    REQUIRE(!colorValue.empty());
+  }
+
+  SECTION("Invalid CSS custom property names are ignored")
+  {
+    // Property names must start with --
+    CSSStyleDeclaration style("-invalid: value; not--valid: value; valid: value;");
+
+    REQUIRE(style.getPropertyValue("-invalid") == "");
+    REQUIRE(style.getPropertyValue("not--valid") == "");
+    REQUIRE(style.getPropertyValue("valid") == "value");
+  }
+
+  SECTION("CSS custom property with fallback value")
+  {
+    CSSStyleDeclaration style("color: var(--undefined-prop, blue);");
+
+    std::string colorValue = style.getPropertyValue("color");
+    REQUIRE(!colorValue.empty());
+  }
+
+  SECTION("CSS custom property inheritance flag")
+  {
+    // CSS custom properties should be marked as inherited
+    REQUIRE(ComputedStyle::IsInheritedProperty("--custom-prop"));
+    REQUIRE(ComputedStyle::IsInheritedProperty("--main-color"));
+    REQUIRE_FALSE(ComputedStyle::IsInheritedProperty("color"));
+    REQUIRE_FALSE(ComputedStyle::IsInheritedProperty("background-color"));
+  }
+}
+
+TEST_CASE("CSS Variables Resolution in ComputedStyle", "[css-variables-resolution]")
+{
+  SECTION("Custom property storage in ComputedStyle")
+  {
+    CSSStyleDeclaration style("--theme-color: #ff0000; --spacing: 10px;");
+    ComputedStyle computedStyle = ComputedStyle::Make(style, nullptr);
+
+    REQUIRE(computedStyle.hasCustomProperty("--theme-color"));
+    REQUIRE(computedStyle.getCustomProperty("--theme-color") == "#ff0000");
+    REQUIRE(computedStyle.hasCustomProperty("--spacing"));
+    REQUIRE(computedStyle.getCustomProperty("--spacing") == "10px");
+    REQUIRE_FALSE(computedStyle.hasCustomProperty("--nonexistent"));
+  }
+
+  SECTION("Variable resolution with defined custom property")
+  {
+    ComputedStyle computedStyle;
+    computedStyle.setCustomProperty("--main-color", "red");
+
+    std::string resolved = computedStyle.resolveVariables("var(--main-color)");
```

Comment:
resolveVariables requires a 2nd argument `computed::Context` which stores the variables.